client/src/app/components/Cart/Cart.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import { cartType } from '../../types/index';
+import ItemInCart from './ItemInCart';
+import styles from './Cart.module.scss';
+
+class Cart extends Component {
+  static propTypes = { cart: cartType.isRequired };
+
+  componentDidMount = () => {
+    window.scrollTo(0, 0);
+  };
+
+  clearCartHandler = e => {
+    e.preventDefault();
+    const { clearCart } = this.props;
+
+    clearCart();
+  };
+
+  render() {
+    const {
+      cart,
+      createNotification,
+      removeFromCart,
+      setColor,
+      setSize,
+      setQuantityAndTotal,
+      setCommonTotal
+    } = this.props;
+    const { productsInCart } = cart;
+    const reflectItems = productsInCart.map(item => (
+      <ItemInCart
+        key={item._id}
+        item={item}
+        createNotification={createNotification}
+        removeFromCart={removeFromCart}
+        setColor={setColor}
+        setSize={setSize}
+        setQuantityAndTotal={setQuantityAndTotal}
+      />
+    ));
+    const styleHead = { borderBottom: 'none' };
+    const total = productsInCart.reduce(
+      (totalSum, el) => totalSum + el.total,
+      0
+    );
+
+    if (cart.total !== total) {
+      setCommonTotal(total);
+    }
+    if (productsInCart.length === 0) {
+      return (
+        <React.Fragment>
+          <div className={styles.empty_cart}>
+            <img
+              src="https://www.qrcardboard.com/images/cart.gif?v=01"
+              alt="Empty cart"
+              className={styles.cart_img}
+            />
+            <p>Your cart is empty! Please order something!</p>
+            <img
```

Comment:
will be great to add the link to PLP here

Reply:
ok, will be done

Reply:
done